src/test/java/seedu/address/logic/commands/BmiCommandTest.java

```diff
@@ -0,0 +1,69 @@
+package seedu.address.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.FitNus;
+import seedu.address.model.ReadOnlyFitNus;
+
+public class BmiCommandTest {
+
+    @Test
+    public void execute_bmiViewedSuccessful() throws Exception {
+        ModelStubWithBmi modelBmiStub = new ModelStubWithBmi();
+
+        CommandResult commandResult = new BmiCommand().execute(modelBmiStub);
+
+        assertEquals(String.format(BmiCommand.MESSAGE_SUCCESS, modelBmiStub.getBmi()),
+                commandResult.getFeedbackToUser());
+
+    }
+
+    @Test
+    public void execute_bmiNotAvailable_throwCommandException() throws Exception {
```

Comment:
Same as comments above. Improve testing by considering more values that BMI could take.